internal/pkg/manifest/backend_svc.go

```diff
@@ -99,6 +101,61 @@ func (s *BackendService) requiredEnvironmentFeatures() []string {
 	return features
 }
 
+// ExposedPorts returns all the ports that are container ports available to receive traffic.
+func (cfg *BackendService) ExposedPorts() ([]ExposedPort, error) {
+	exposedPorts := make(map[int]ExposedPort)
```

Comment:
I don't quite follow why we need the map, can't we just use the slice?

I see that we have validation that ensures that no two ports can be declared twice in the manifest, due to that can this method just assume that ports are unique?

Reply:
So the reason why I took this as map because we are writing some logic here that determines the correct set of exposed ports for example, if `target_port` is given and `target_container` is nil then we try our best to figure out the `target_container` from the `target_port`. In that case, in the for loop of sidecars  if `target_port` is same as that of the sidecar port then we set the exposedPorts[target_port] to sidecar container which was earlier set as `exposedPorts[target_port] = primary container`. That is how we are making a use of the map here. Else we will have to write extra logic to avoid duplication of the exposedPorts entry. Does that make sense?